### PR TITLE
Ro sent

### PR DESCRIPTION
**Pull request by iliemihai**
Movies reviews dataset for Romanian language.

### CODE REVIEW

**Comment by SBrandeis** [comment]
Hi @iliemihai, it looks like this PR holds changes from your previous PR #1493 .
Would you mind removing them from the branch please ?

File: datasets/rosent/rosent.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        ROSENTConfig(name=DEFAULT_CONFIG_NAME, version=VERSION, description="Romanian sentiment dataset."),
+    ]
+
+    def _info(self):
+
+        features = datasets.Features(
+            {
+                "id": datasets.Value("string"),
+                "sentence": datasets.Sequence(datasets.Value("string")),
+                "label": datasets.Sequence(datasets.Value("int32")),
```
Comment (SBrandeis) [inline datasets/rosent/rosent.py:66]: Looks like this could be a `ClassLabel` feature

Comment (SBrandeis) [inline datasets/rosent/rosent.py:66]: Why are these two features Sequences ? It looks like each id has 1 sentence and its associated label

File: datasets/rosent/rosent.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+
+        urls_to_download_train = _URL + _TRAINING_FILE
+        urls_to_download_test = _URL + _TEST_FILE
+
+        train_path = dl_manager.download(urls_to_download_train)
+        test_path = dl_manager.download(urls_to_download_test)
```
Comment (SBrandeis) [inline datasets/rosent/rosent.py:94]: I'd recommend using this syntax, like you did in your previous PR:
```suggestion
        urls = {
            "train": _URL + _TRAINING_FILE,
            "test": _URL + _TEST_FILE,
        }
        paths = dl_manager.download(urls)
```

As it enables parrallelism in the download-extract task

File: datasets/rosent/rosent.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        print("FISIERE LUATE", train_path, test_path)
```
Comment (SBrandeis) [inline datasets/rosent/rosent.py:95]: Don't forget to remove debugging assets :)
```suggestion
```

File: datasets/rosent/rosent.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        """ Yields examples. """
+
+        logging.info("⏳ Generating examples from = %s", filepath)
+        with open(filepath, encoding="utf-8") as f:
+            # data = pd.read_csv(filepath)
+            data = csv.reader(f, delimiter=",", quotechar='"')
+
+            next(data, None)
+            for row_id, row in enumerate(data):
+                print("ROW", row)
```
Comment (SBrandeis) [inline datasets/rosent/rosent.py:120]:
```suggestion
```

File: datasets/rosent/rosent.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                id, txt, lbl = row
+                yield "{}_{}".format(row_id, id), {
+                    "id": id,
+                    "sentence": [txt],
+                    "label": [lbl],
```
Comment (SBrandeis) [inline datasets/rosent/rosent.py:125]: Provided you change the features declaration, there's no need to encapsulate this in a list:
```suggestion
                    "sentence": txt,
                    "label": lbl,
```

File: datasets/rosent/rosent.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                gen_kwargs={"filepath": test_path},
+            ),
+        ]
+
+    def _generate_examples(self, filepath):
+        """ Yields examples. """
+
+        logging.info("⏳ Generating examples from = %s", filepath)
+        with open(filepath, encoding="utf-8") as f:
+            # data = pd.read_csv(filepath)
```
Comment (SBrandeis) [inline datasets/rosent/rosent.py:115]:
```suggestion
```

File: datasets/rosent/rosent.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+            data = csv.reader(f, delimiter=",", quotechar='"')
+
+            next(data, None)
+            for row_id, row in enumerate(data):
```
Comment (SBrandeis) [inline datasets/rosent/rosent.py:119]: You might want to have a look at the [csv.DictReader](https://docs.python.org/fr/3.6/library/csv.html#csv.DictReader) object

It takes cares of the header line for you

File: datasets/rosent/rosent.py [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
```
Comment (SBrandeis) [inline datasets/rosent/rosent.py:78]: I think it makes sens to have supervised keys here:
```suggestion
            supervised_keys=("sentence", "label"),
```

**Review by SBrandeis — CHANGES_REQUESTED** [review]
Very cool, thanks a lot !

I have a few comments regarding the script 👇

You will need to re-generate the `dataset_infos.json` file after making changes to the fetaures declaration.

Otherwise, the dataset card is missing (the `README.md` file). You can find a template and a guid on how to complete it under the `templates` directory at the root of the library. You may run the [`datasets-tagging`](https://github.com/huggingface/datasets-tagging) app to generate the YAML tags.

Feel free to reach out if you have any questions !

**Comment by iliemihai** [comment]
@SBrandeis I am sorry. Yes I will remove them. Thank you :D

File: datasets/rosent/rosent.py (hunk 1 above)
Comment (iliemihai) [reply datasets/rosent/rosent.py:66]: I was thinking about "label": datasets.Sequence(datasets.Value("int32")), as being 0 or 1 for positive or negative sentiment

Comment (iliemihai) [reply datasets/rosent/rosent.py:66]: Ah yes, it should not be sequence : `"label": datasets.Sequence(datasets.Value("int32")),` -> `"label": datasets.Value("int32"),` :D

Comment (lhoestq) [reply datasets/rosent/rosent.py:66]: ClassLabel is useful to add names to integer (often binary) values.
Here you can use for example
```
"label": datasets.Sequence(datasets.ClassLabel(names=["negative", "positive"]))
```
where `negative` and `positive` are meaningful labels for sentiment analysis.

In practice the values that are stored will still be 0 and 1, but we'll know their string representation as well.

Comment (lhoestq) [reply datasets/rosent/rosent.py:66]: Or actually

```
"label": datasets.ClassLabel(names=["negative", "positive"])
```

;)

**Review by lhoestq — COMMENTED** [review]
Looking good so far :)
Please don't forget to add the dataset cards with the YAML tags (more infos [here](https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#tag-the-dataset-and-write-the-dataset-card))

**Comment by gchhablani** [comment]
Hi @lhoestq @SBrandeis @iliemihai

Is this still in progress or can I take over this one?

Thanks,
Gunjan

**Comment by gchhablani** [comment]
Hi,
While trying to add this dataset, I found some potential issues. 
The homepage mentioned is : https://github.com/katakonst/sentiment-analysis-tensorflow/tree/master/datasets/ro/, where the dataset is different from the URLs: https://raw.githubusercontent.com/dumitrescustefan/Romanian-Transformers/examples/examples/sentiment_analysis/ro/train.csv. It is unclear which dataset is "correct". I checked the total examples (train+test) in both places and they do not match.

**Comment by lhoestq** [comment]
We should use the data from dumitrescustefan and set the homepage to his repo IMO, since he's first author of the paper of the dataset.

**Comment by gchhablani** [comment]
Hi @lhoestq,

Well, that makes sense. But, while the dataset link works, it is not present on their repo and I don't understand how or why that is the case.

**Comment by gchhablani** [comment]
Hi @lhoestq,

Cool, I'll get working on it.

Thanks

**Comment by gchhablani** [comment]
Hi @lhoestq, 

This PR can be closed.

**Comment by lhoestq** [comment]
Closing in favor of #2011 
Thanks again for adding it !
